yt/frontends/stream/tests/test_stream_particles.py

```diff
@@ -218,6 +218,53 @@ def test_load_particles_types():
         assert dd["all", f"particle_position_{ax}"].size == num_tot_particles
 
 
+def test_load_particles_sph_types():
+    num_particles = 10000
+
+    data = {
+        ("gas", "particle_position_x"): np.random.random(size=num_particles),
+        ("gas", "particle_position_y"): np.random.random(size=num_particles),
+        ("gas", "particle_position_z"): np.random.random(size=num_particles),
+        ("gas", "particle_velocity_x"): np.random.random(size=num_particles),
+        ("gas", "particle_velocity_y"): np.random.random(size=num_particles),
+        ("gas", "particle_velocity_z"): np.random.random(size=num_particles),
+        ("gas", "particle_mass"): np.ones(num_particles),
+        ("gas", "density"): np.ones(num_particles),
+        ("gas", "smoothing_length"): np.ones(num_particles),
+        ("dm", "particle_position_x"): np.random.random(size=num_particles),
+        ("dm", "particle_position_y"): np.random.random(size=num_particles),
+        ("dm", "particle_position_z"): np.random.random(size=num_particles),
+        ("dm", "particle_velocity_x"): np.random.random(size=num_particles),
+        ("dm", "particle_velocity_y"): np.random.random(size=num_particles),
+        ("dm", "particle_velocity_z"): np.random.random(size=num_particles),
+        ("dm", "particle_mass"): np.ones(num_particles),
+    }
+
+    ds = load_particles(data)
+
+    assert set(ds.particle_types) == {"gas", "dm"}
```

Comment:
I wondered for a minute why the attribute was converted to `set` here, I assume it's because we don't care about order ? Also, do we actually care about exact set equality ?
In any case, I suggest adding a comment.

Reply:
We don't care about order.  I think this is OK as it is, especially since we really do want to make sure that the sets are equal, and not just that all of them are included, etc.

Reply:
I wasn't arguing against it, I just wanted to point out that a comment might help make intentions clearer